src/efftester.ml

```diff
@@ -1089,6 +1185,51 @@ module GeneratorsWithContext (Ctx : Context) = struct
     in
     [ (3, gen) ]
 
+  and option_intro_rules env t eff size =
+    let gen =
+      let open Gen in
+      match t with
+      | Option t' ->
+        Gen.frequencyl [ (2, "Some"); (1, "None") ] >>= fun name ->
+        (match name with
+        | "None" -> return (Some (none t))
+        | "Some" ->
+          list_permute_term_gen_outer env t' eff (size / 3) >>= fun trm ->
+          (match trm with
+          | None -> return None
+          | Some trm' -> return (Some (some t trm')))
+        | _ -> failwith "option_intro_rules: impossible option adt_constr name")
+      | _ -> return None
+    in
+    [ (100, gen) ]
```

Comment:
Why use `100` as the weight? This sounds like a lot compared to the other rules. Was it a choice for debugging, or is it a deliberate choice to generate constructors more often when an `option` is expected?